src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/ExcGemini.java

```diff
@@ -0,0 +1,152 @@
+package com.pirimid.cryptotrade.helper.exchange.gemini;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.pirimid.cryptotrade.DTO.PlaceOrderReqDTO;
+import com.pirimid.cryptotrade.DTO.PlaceOrderResDTO;
+import com.pirimid.cryptotrade.DTO.SymbolResDTO;
+import com.pirimid.cryptotrade.helper.exchange.ExcParent;
+import com.pirimid.cryptotrade.helper.exchange.gemini.dto.request.CreateOrderRequest;
+import com.pirimid.cryptotrade.helper.exchange.gemini.dto.response.CancelOrderResponse;
+import com.pirimid.cryptotrade.helper.exchange.gemini.dto.response.CreateOrderResponse;
+import com.pirimid.cryptotrade.helper.exchange.gemini.dto.response.SymbolResponse;
+import com.pirimid.cryptotrade.util.GeminiUtil;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.*;
+
+@Component
+public class ExcGemini implements ExcParent {
+
+    @Value("${api.exchange.gemini.baseurl}")
+    private String baseUrl;
+    private long getNonce(){
+        return new Date().getTime();
+    }
+    private byte[] getB64(String payload){
+        return Base64.getEncoder().encode(payload.getBytes(StandardCharsets.UTF_8));
+    }
+    private ResponseEntity<String> apiCaller(String uri,String reqType) throws IOException, InterruptedException {
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(uri))
+                .setHeader("X-GEMINI-APIKEY","master-sIFFhvWpYBof8QH2XECh")
```

Comment:
why fixed api key?

Reply:
removed, was not necessary for open endpoints